skrub/_gap_encoder.py

```diff
@@ -467,7 +467,7 @@ def fit(self, X, y=None):
         unq_X, unq_V, lookup = self._init_vars(X, is_null)
         n_batch = (len(X) - 1) // self.batch_size + 1
         n_samples = len(X)
-        del X
```

Comment:
Can we not get this "del X" back?

It would help decreasing memory pressure

Reply:
Indeed, I forgot